haystack/errors.py

```diff
@@ -13,6 +14,7 @@ class HaystackError(Exception):
     `HaystackError.message` will exist and have the expected content.
     """
 
+    @send_event
```

Comment:
Thanks for the feedback. I refactored `errors.py` so that we get a single `PipelineSchemaError raised` event by only sending an event within `HaystackError`. The implementation is the same as for `BaseComponent`.